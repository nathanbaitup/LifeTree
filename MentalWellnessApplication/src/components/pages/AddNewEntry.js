import React, { useState, useEffect } from 'react';
import { Text, View, StyleSheet, ImageBackground, TextInput, TouchableOpacity, Image, Keyboard } from 'react-native';

// Imports firestore from firebase to save user entries to the firstore database.
import firestore from '@react-native-firebase/firestore';

export default function AddNewEntry(props) {

    // Initializing the state so that when a user selects a mood, it is outlined to show they have selected it.
    const [angry, setAngry] = useState(false);
    const [sad, setSad] = useState(false);
    const [meh, setMeh] = useState(false);
    const [happy, setHappy] = useState(false);

    // Initializing the state so that once a user has entered all of their details it can be stored to the firestore database.
    const [journalEntry, setJournalEntry] = useState('');
    const [usefulDate, setUsefulDate] = useState(null);
    const [displayDate, setDisplayDate] = useState(null);
    const [entryMood, setEntryMood] = useState('');

    // Creates a reference to the journal list collection in firestore to save data.
    const journalsRef = firestore().collection('journalList');
    // Gets the users ID from props passed in from App.js.
    const userID = props.extraData.id;

    useEffect(() => {
        // Gets the current date and creates an object on how to display the date.
        const date = new Date();
        const displayOptions = { day: 'numeric', month: 'long', year: 'numeric' };

        // Displays the date on the component in a nice format.
        const displayDateOfEntry = date.toLocaleDateString('en-US', displayOptions);
        setDisplayDate(displayDateOfEntry);

        // Takes the date in ISO format to be saved to the firestore databse and be displayed on the mood calendar.
        const usefulDateOfEntry = date.toISOString().split('T')[0];
        setUsefulDate(usefulDateOfEntry);

    }, []);


    // On submission, checks if there is an input for journal entry, then creates a timestamp and collects the userID,
    // mood, date, and jorunal entry to be stored in the firestore databse, then clears the fields and notifies the user if
    // the upload was successful or if an error occurs.
    const onSubmitButtonPress = () => {
        // REFERENCE ACCESSED 14/12/2021 https://www.freecodecamp.org/news/react-native-firebase-tutorial/ (8. Writing and Reading Data from Firestore)
        // Used to learn how to save data to firestore.
        if (journalEntry && journalEntry.length > 0) {
            const timestamp = firestore.FieldValue.serverTimestamp();
            const data = {
                authorID: userID,
                moodSelected: entryMood,
                journalText: journalEntry,
                moodCalendarDate: usefulDate,
                dateOfEntry : displayDate,
                createdAt: timestamp
            };
            journalsRef
                .add(data)
                .then((_doc) => {
                    Keyboard.dismiss();

                    setJournalEntry('');
                    setEntryMood('');
                    setAngry(false);
                    setSad(false);
                    setMeh(false);
                    setHappy(false);

                    alert('Entry Successfully Added');

                })
                .catch((error) => {
                    alert('Error: ' + error);
                });
        }
        //END REFERENCE
    };

    // ---------- Functions to set the user mood ----------
    // Used to highlight the users mood as well as set the mood to the state.
    const isAngry = () => {
        if (!angry) {
            setAngry(true);
            setSad(false);
            setMeh(false);
            setHappy(false);
            setEntryMood('Angry');
        }
    };
    const isSad = () => {
        if (!sad) {
            setAngry(false);
            setSad(true);
            setMeh(false);
            setHappy(false);
            setEntryMood('Sad');
        }
    };
    const isMeh = () => {
        if (!meh) {
            setAngry(false);
            setSad(false);
            setMeh(true);
            setHappy(false);
            setEntryMood('Meh');
        }
    };
    const isHappy = () => {
        if (!happy) {
            setAngry(false);
            setSad(false);
            setMeh(false);
            setHappy(true);
            setEntryMood('Happy');
        }
    };

    return (
        <ImageBackground source={require('../resources/img/background.png')} style={{ width: '100%', height: '100%', opacity: 50 }} >
            <View style={styles.mainContainer}>
                <View style={styles.contentContainer}>
                    <Text style={styles.header}>How are you feeling today?</Text>

                    <View style={styles.moodModules}>

                        <TouchableOpacity style={angry ? styles.moodModSelected : styles.moodModUnselected} onPress={isAngry} >
                            <Image source={require('../resources/img/faces/angry.png')} style={styles.moodFaces} />
                            <Text>Angry</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={sad ? styles.moodModSelected : styles.moodModUnselected} onPress={isSad}>
                            <Image source={require('../resources/img/faces/sad.png')} style={styles.moodFaces} />
                            <Text>Sad</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={meh ? styles.moodModSelected : styles.moodModUnselected} onPress={isMeh}>
                            <Image source={require('../resources/img/faces/meh.png')} style={styles.moodFaces} />
                            <Text>Meh</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={happy ? styles.moodModSelected : styles.moodModUnselected} onPress={isHappy}>
                            <Image source={require('../resources/img/faces/happy.png')} style={styles.moodFaces} />
                            <Text>Happy</Text>

                        </TouchableOpacity>
                    </View>

                    <View>
                        <Text style={styles.journalHeader}>Any comments for the day? </Text>
                        <TextInput style={styles.journalEntry}
                            placeholder='Feel free to dump as much or as little information in here as you want. We wont judge you. We promise! '
                            numberOfLines={10}
                            multiline={true}
                            onChangeText={(text) => setJournalEntry(text)}
                            value={journalEntry}
                        />

                        <Text style={styles.date}>Todays Date: {displayDate} </Text>
                        <View style={styles.submitButtonContainer}>

                            {/* TODO: MAKE ONPRESS TO SAVE DATA WHEN BACKEND HAS BEEN CONNECTED */}
                            <TouchableOpacity style={styles.submitButton} onPress={onSubmitButtonPress} >
                                <Text style={styles.submitText}>Submit</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
            </View>
        </ImageBackground>
    );
}

// Styling for the document.
const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        justifyContent: 'center',
    },
    contentContainer: {
        margin: 20,
        padding: 10,
        backgroundColor: '#FFFFFF',
        borderRadius: 25,
    },
    header: {
        fontSize: 20,
        fontWeight: 'bold',
        color: '#000000'
    },
    date: {
        fontSize: 16,
        fontWeight: 'bold',
        color: '#000000'
    },
    moodModules: {
        flexDirection: 'row',
        paddingBottom: 50,
    },
    moodModUnselected: {
        marginTop: 10,
        marginLeft: 5,
        textAlign: 'center',
        paddingLeft: 10,
    },
    moodModSelected: {
        marginTop: 10,
        marginLeft: 5,
        textAlign: 'center',
        borderWidth: 1,
        borderColor: '#000000',
        borderRadius: 5,
        paddingLeft: 10,
    },
    moodFaces: {
        marginTop: 10,
        height: 50,
        width: 66,
        alignItems: 'center',
    },
    journalEntry: {
        marginTop: 5,
        marginBottom: 20,
        backgroundColor: '#ebebeb',
        borderRadius: 10,
        maxHeight: 250,
        height: 150,
    },
    journalHeader: {
        fontSize: 16,
        fontStyle: 'italic',
        fontWeight: 'bold',
        color: '#000000',
    },
    submitButtonContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        paddingTop: 20,
    },
    submitButton: {
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#00e676',
        borderRadius: 15,
        width: 150,
    },
    submitText: {
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 18,
        color: '#000000',
    },
});
