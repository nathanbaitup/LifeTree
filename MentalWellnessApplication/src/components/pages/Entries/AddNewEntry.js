import React, { useState, useEffect } from 'react';
import { Text, View, ImageBackground, TextInput, TouchableOpacity, Image, Keyboard } from 'react-native';

// Imports the documents styling.
import { entryStyles } from './Styles';

// Imports firestore from firebase to save user entries to the firstore database.
import firestore from '@react-native-firebase/firestore';

export default function AddNewEntry(props) {

    // Initializing the state so that when a user selects a mood, it is outlined to show they have selected it.
    const [angry, setAngry] = useState(false);
    const [sad, setSad] = useState(false);
    const [meh, setMeh] = useState(false);
    const [happy, setHappy] = useState(false);

    // Initializing the state so that once a user has entered all of their details it can be stored to the firestore database.
    const [journalEntry, setJournalEntry] = useState('');
    const [usefulDate, setUsefulDate] = useState(null);
    const [displayDate, setDisplayDate] = useState(null);
    const [entryMood, setEntryMood] = useState('');
    const [obsession, setObsession] = useState('');

    // Creates a reference to the journal list collection in firestore to save data.
    const journalsRef = firestore().collection('journalList');
    // Gets the users ID from props passed in from App.js.
    const userID = props.extraData.id;

    useEffect(() => {
        // Gets the current date and creates an object on how to display the date.
        const date = new Date();
        const displayOptions = { day: 'numeric', month: 'long', year: 'numeric' };

        // Displays the date on the component in a nice format.
        const displayDateOfEntry = date.toLocaleDateString('en-US', displayOptions);
        setDisplayDate(displayDateOfEntry);

        // Takes the date in ISO format to be saved to the firestore databse and be displayed on the mood calendar.
        const usefulDateOfEntry = date.toISOString().split('T')[0];
        setUsefulDate(usefulDateOfEntry);

    }, []);


    // On submission, checks if there is an input for journal entry, then creates a timestamp and collects the userID,
    // mood, date, and jorunal entry to be stored in the firestore databse, then clears the fields and notifies the user if
    // the upload was successful or if an error occurs.
    const onSubmitButtonPress = () => {
        // REFERENCE ACCESSED 14/12/2021 https://www.freecodecamp.org/news/react-native-firebase-tutorial/ (8. Writing and Reading Data from Firestore)
        // Used to learn how to save data to firestore.
        if (journalEntry && journalEntry.length > 0) {
            const timestamp = firestore.FieldValue.serverTimestamp();
            const data = {
                authorID: userID,
                moodSelected: entryMood,
                journalText: journalEntry,
                obsessionText: obsession,
                moodCalendarDate: usefulDate,
                dateOfEntry: displayDate,
                createdAt: timestamp
            };
            journalsRef
                .add(data)
                .then(() => {
                    Keyboard.dismiss();

                    setJournalEntry('');
                    setEntryMood('');
                    setObsession('');
                    setAngry(false);
                    setSad(false);
                    setMeh(false);
                    setHappy(false);

                    alert('Entry Successfully Added');

                })
                .catch((error) => {
                    alert('Error: ' + error);
                });
        }
        //END REFERENCE
    };

    // ---------- Functions to set the user mood ----------
    // Used to highlight the users mood as well as set the mood to the state.
    const isAngry = () => {
        if (!angry) {
            setAngry(true);
            setSad(false);
            setMeh(false);
            setHappy(false);
            setEntryMood('Angry');
        }
    };
    const isSad = () => {
        if (!sad) {
            setAngry(false);
            setSad(true);
            setMeh(false);
            setHappy(false);
            setEntryMood('Sad');
        }
    };
    const isMeh = () => {
        if (!meh) {
            setAngry(false);
            setSad(false);
            setMeh(true);
            setHappy(false);
            setEntryMood('Meh');
        }
    };
    const isHappy = () => {
        if (!happy) {
            setAngry(false);
            setSad(false);
            setMeh(false);
            setHappy(true);
            setEntryMood('Happy');
        }
    };

    return (
        <ImageBackground source={require('../../../resources/img/background.png')} style={{ width: '100%', height: '100%', opacity: 50 }} >
            <View style={entryStyles.mainContainer}>
                <View style={entryStyles.contentContainer}>
                    <Text style={entryStyles.header}>How are you feeling today?</Text>

                    <View style={entryStyles.moodModules}>

                        <TouchableOpacity style={angry ? entryStyles.moodModSelected : entryStyles.moodModUnselected} onPress={isAngry} >
                            <Image source={require('../../../resources/img/faces/angry.png')} style={entryStyles.moodFaces} />
                            <Text>Angry</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={sad ? entryStyles.moodModSelected : entryStyles.moodModUnselected} onPress={isSad}>
                            <Image source={require('../../../resources/img/faces/sad.png')} style={entryStyles.moodFaces} />
                            <Text>Sad</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={meh ? entryStyles.moodModSelected : entryStyles.moodModUnselected} onPress={isMeh}>
                            <Image source={require('../../../resources/img/faces/meh.png')} style={entryStyles.moodFaces} />
                            <Text>Meh</Text>
                        </TouchableOpacity>

                        <TouchableOpacity style={happy ? entryStyles.moodModSelected : entryStyles.moodModUnselected} onPress={isHappy}>
                            <Image source={require('../../../resources/img/faces/happy.png')} style={entryStyles.moodFaces} />
                            <Text>Happy</Text>

                        </TouchableOpacity>
                    </View>

                    <View>
                        <Text style={entryStyles.subHeader}>Any comments for the day? </Text>
                        <TextInput style={entryStyles.journalEntry}
                            placeholder='Feel free to dump as much or as little information in here as you want. We wont judge you. We promise! '
                            numberOfLines={10}
                            multiline={true}
                            onChangeText={(text) => setJournalEntry(text)}
                            value={journalEntry}
                        />
                        <Text style={entryStyles.subHeader}>Obsession of the day?</Text>
                        <TextInput style={entryStyles.obsessionEntry}
                            placeholder='Enter your random obsession here!'
                            numberOfLines={1}
                            multiline={true}
                            onChangeText={(text) => setObsession(text)}
                            value={obsession}
                        />
                        <Text style={entryStyles.date}>Todays Date: {displayDate} </Text>
                        <View style={entryStyles.submitButtonContainer}>
                            <TouchableOpacity style={entryStyles.submitButton} onPress={onSubmitButtonPress} >
                                <Text style={entryStyles.submitText}>Submit</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
            </View>
        </ImageBackground>
    );
}

