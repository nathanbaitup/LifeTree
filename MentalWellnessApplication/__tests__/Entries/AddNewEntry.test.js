// https://keyholesoftware.com/2020/11/02/using-jest-and-testing-library-with-react-native-part-1-setting-up/
// helped with setting up testing.

import 'react-native';
import React from 'react';
import AddNewEntry from '../../src/components/pages/Entries/AddNewEntry';

import { cleanup, render, fireEvent } from '@testing-library/react-native';

afterEach(cleanup);

describe('Check that the add new entry page is rendered and elements are displayed and updateable.', () => {
  const extraData = { 'id': 1, 'fullName': 'John Doe' };
  const addNewEntryRender = render(<AddNewEntry  extraData={extraData} />).toJSON();
  
  
  it('Checks to see if the Home page renders', () => {
      expect(addNewEntryRender).toMatchSnapshot();
  });

  it('Checks to see if the moods are displayed.', () => {
    const {getByTestId} = render(<AddNewEntry extraData={extraData} />);
    const happyRender = getByTestId('happy');
    const mehRender = getByTestId('meh');
    const sadRender = getByTestId('sad');
    const angryRender = getByTestId('angry');

    expect(happyRender.type).toBe('Image');
    expect(mehRender.type).toBe('Image');
    expect(sadRender.type).toBe('Image');
    expect(angryRender.type).toBe('Image');
    
  });

  it('Checks to see if the journal and obsession inputs are displayed.', () => {
    const {getByTestId} = render(<AddNewEntry extraData={extraData} />);

    const journalID = getByTestId('journalInput');
    const obsessionID = getByTestId('obsessionInput');

    const journalPlaceholder = 'Feel free to dump as much or as little information in here as you want. We wont judge you. We promise! ';
    const obsessionPlaceholder = 'Enter your random obsession here!';
    
    expect(journalID.props.placeholder).toBe(journalPlaceholder);
    expect(obsessionID.props.placeholder).toBe(obsessionPlaceholder);


  });

  it('Checks to see if the date is displayed and set to the current date.', () => {
    const {getByTestId} = render(<AddNewEntry extraData={extraData} />);
    const date = new Date();
    const displayDate = date.toLocaleDateString('en-US', { day: 'numeric', month: 'long', year: 'numeric'});
    const dateRendered = getByTestId('dateID');

    expect(dateRendered.props.children[1]).toBe(displayDate);
  });

  it('Checks to see if the submit button is accessible via accessibility label and ID.', () => {
    const {getByTestId, getByA11yLabel} = render(<AddNewEntry extraData={extraData} />);
    const submitButton = getByA11yLabel('Submit Button');
    const buttonID = getByTestId('submitBTN');

    expect(submitButton).toBeTruthy();
    expect(buttonID).toBeTruthy();
  });

  it('Update the journal entry and obsession entry to contain text.', () => {
    const {getByTestId} = render(<AddNewEntry extraData={extraData} />);
    const journalInput = getByTestId('journalInput');
    const obsessionInput = getByTestId('obsessionInput');
    // Expect a blank entry when first loading the page.
    expect(journalInput.props.value).toEqual('');
    expect(obsessionInput.props.value).toEqual('');

    // Add values to inputs.
    fireEvent.changeText(journalInput, 'Test journal entry');
    fireEvent.changeText(obsessionInput, 'Test obsession entry');

    // Expect values to be stored in the inputs.
    expect(journalInput.props.value).toEqual('Test journal entry');
    expect(obsessionInput.props.value).toEqual('Test obsession entry');
  });

});
