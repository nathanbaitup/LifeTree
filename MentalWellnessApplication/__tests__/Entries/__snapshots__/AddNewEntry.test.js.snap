// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that the add new entry page is rendered and elements are displayed and updateable. Checks to see if the Home page renders 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 25,
          "margin": 20,
          "padding": 10,
        }
      }
    >
      <RCTScrollView>
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            How are you feeling today?
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingBottom": 50,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": 5,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingLeft": 10,
                  "textAlign": "center",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/img/faces/angry.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "marginTop": 10,
                    "width": 66,
                  }
                }
                testID="angry"
              />
              <Text>
                Angry
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": 5,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingLeft": 10,
                  "textAlign": "center",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/img/faces/sad.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "marginTop": 10,
                    "width": 66,
                  }
                }
                testID="sad"
              />
              <Text>
                Sad
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": 5,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingLeft": 10,
                  "textAlign": "center",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/img/faces/meh.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "marginTop": 10,
                    "width": 66,
                  }
                }
                testID="meh"
              />
              <Text>
                Meh
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": 5,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingLeft": 10,
                  "textAlign": "center",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/img/faces/happy.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 50,
                    "marginTop": 10,
                    "width": 66,
                  }
                }
                testID="happy"
              />
              <Text>
                Happy
              </Text>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                }
              }
            >
              Any comments for the day? 
            </Text>
            <TextInput
              multiline={true}
              numberOfLines={10}
              onChangeText={[Function]}
              placeholder="Feel free to dump as much or as little information in here as you want. We wont judge you. We promise! "
              style={
                Object {
                  "backgroundColor": "#ebebeb",
                  "borderRadius": 10,
                  "height": 150,
                  "marginBottom": 20,
                  "marginTop": 5,
                  "maxHeight": 250,
                }
              }
              testID="journalInput"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                }
              }
            >
              Obsession of the day?
            </Text>
            <TextInput
              multiline={true}
              numberOfLines={1}
              onChangeText={[Function]}
              placeholder="Enter your random obsession here!"
              style={
                Object {
                  "backgroundColor": "#ebebeb",
                  "borderRadius": 10,
                  "height": 40,
                  "marginBottom": 20,
                  "marginTop": 5,
                  "maxHeight": 40,
                }
              }
              testID="obsessionInput"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
              testID="dateID"
            >
              Todays Date: 
              January 9, 2022
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingTop": 20,
                  "textAlign": "center",
                }
              }
            >
              <View
                accessibilityLabel="Submit Button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#00e676",
                    "borderRadius": 15,
                    "justifyContent": "center",
                    "opacity": 1,
                    "padding": 15,
                    "width": 150,
                  }
                }
                testID="submitBTN"
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
